include/test_cases.hpp

```diff
@@ -0,0 +1,251 @@
+#ifndef TEST_CASES_HPP
+#define TEST_CASES_HPP
+
+#include <deal.II/base/function.h>
+#include <deal.II/base/point.h>
+#include <cmath>
+
+using namespace dealii;
+
+namespace WaveEquationProject {
+
+
+    // =========================================================================
+    // SCENARIO 1: MANUFACTURED SOLUTION (Dim-Independent)
+    // =========================================================================
+    // Description:
+    // Uses a smooth analytical function to verify the order of convergence.
+    // The solution is defined as a standing wave product of sines:
+    // u(x,t) = sin(pi*t) * product(sin(pi*x_i)) for i=0..dim-1
+    //
+    // In 2D: u = sin(pi*t) * sin(pi*x) * sin(pi*y)
+    // In 3D: u = sin(pi*t) * sin(pi*x) * sin(pi*y) * sin(pi*z)
+    // =========================================================================
+    
+    /**
+     * @brief Exact analytical solution u_ex(x,t).
+     */
+    template <int dim>
+    class ExactSolution_Case1 : public Function<dim> {
+    public:
+        ExactSolution_Case1() : Function<dim>(1) {} 
+        
+        virtual double value(const Point<dim> &p, const unsigned int /*component*/ = 0) const override {
+            double time = this->get_time();
```

Comment:
```suggestion
            const double time = this->get_time();
```
If something is not supposed to change, you should always declare it as `const`. This helps a lot with compiler optimization, which in many cases leads to measurably better performance. This is probably not one of those cases, but you should get in the habit of using `const` appropriately as part of your "coding hygiene".